yt_dlp/extractor/rutv.py

```diff
@@ -192,13 +191,15 @@ def _real_extract(self, url):
                         'url': url
                     }
                 fmt.update({
-                    'width': width,
-                    'height': height,
+                    'width': int_or_none(quality, default=height, invscale=width, scale=height),
+                    'height': int_or_none(quality, default=height),
                     'format_id': '%s-%s' % (transport, quality),
+                    'quality': quality,
+                    'preference': preference,
                 })
                 formats.append(fmt)
 
-        self._sort_formats(formats)
+        self._sort_formats(formats, ('quality', 'preference'))
```

Comment:
```py
                    'source_preference': preference,
                })
                formats.append(fmt)

        self._sort_formats(formats)
```
The video is geo-blocked for me, but I'm pretty sure this is what u want.

There is no point in setting `quality`=`height`

Reply:
`quality` seems to be height actually for HTTP formats though, so that it can calculate resolution

Reply:
I meant there is no pint in setting the field `quality` the same as what we are setting `height`. Better to set only `height`

Reply:
also, quality should be a number

Reply:
oh okay
makes sense